# 01 Import Library

In [62]:
# Import libraries
import pandas as pd
import numpy as np
import os


# 02 Import Data

In [97]:
#import movie_datase.csv
df_movie_dataset = pd.read_csv(r'/Users/elenasharavara/Documents/movie_dataset.csv',index_col = False)

In [98]:
path = '/Users/elenasharavara/Documents'

# 03 Data Cleaning and Wrangling

In [99]:
df_movie_dataset.shape

(934745, 23)

In [100]:
df_movie_dataset.head()

id            title  vote_average  vote_count    status release_date  \
0   27205        Inception         8.364       34495  Released   2010-07-15   
1  157336     Interstellar         8.417       32571  Released   2014-11-05   
2     155  The Dark Knight         8.512       30619  Released   2008-07-16   
3   19995           Avatar         7.573       29815  Released   2009-12-15   
4   24428     The Avengers         7.710       29166  Released   2012-04-25   

      revenue  runtime  adult                     backdrop_path  ...  \
0   825532764      148  False  /8ZTVqvKDQ8emSGUEMjsS4yHAwrp.jpg  ...   
1   701729206      169  False  /pbrkL804c8yAv3zBZR4QPEafpAR.jpg  ...   
2  1004558444      152  False  /nMKdUUepR0i5zn0y1T4CsSB5chy.jpg  ...   
3  2923706026      162  False  /vL5LR6WdxWPjLPFRLe133jXWsh5.jpg  ...   
4  1518815515      143  False  /9BBTo63ANSmhC4e6r62OJFuK2GL.jpg  ...   

   original_language   original_title  \
0                 en        Inception   
1                 en     Interstellar   
2                 en  The Dark Knight   
3                 en           Avatar   
4                 en     The Avengers   

                                            overview popularity  \
0  Cobb, a skilled thief who commits corporate es...     83.952   
1  The adventures of a group of explorers who mak...    140.241   
2  Batman raises the stakes in his war on crime. ...    130.643   
3  In the 22nd century, a paraplegic Marine is di...     79.932   
4  When an unexpected enemy emerges and threatens...     98.082   

                        poster_path  \
0  /oYuLEt3zVCKq57qu2F8dT7NIa6f.jpg   
1  /gEU2QniE6E77NI6lCU6MxlNBvIx.jpg   
2  /qJ2tW6WMUDux911r6m7haRef0WH.jpg   
3  /kyeqWdyUXW608qlYkRqosgbbJyK.jpg   
4   /RYMX2wcKCBAr24UyPD7xwmjaTn.jpg   

                                             tagline  \
0               Your mind is the scene of the crime.   
1  Mankind was born on Earth. It was never meant ...   
2                  Welcome to a world without rules.   
3                        Enter the world of Pandora.   
4                            Some assembly required.   

                                        genres  \
0           Action, Science Fiction, Adventure   
1            Adventure, Drama, Science Fiction   
2               Drama, Action, Crime, Thriller   
3  Action, Adventure, Fantasy, Science Fiction   
4           Science Fiction, Action, Adventure   

                                production_companies  \
0  Legendary Pictures, Syncopy, Warner Bros. Pict...   
1  Legendary Pictures, Syncopy, Lynda Obst Produc...   
2  DC Comics, Legendary Pictures, Syncopy, Isobel...   
3  Dune Entertainment, Lightstorm Entertainment, ...   
4                                     Marvel Studios   

                       production_countries  \
0  United Kingdom, United States of America   
1  United Kingdom, United States of America   
2  United Kingdom, United States of America   
3  United States of America, United Kingdom   
4                  United States of America   

                     spoken_languages  
0  English, French, Japanese, Swahili  
1                             English  
2                   English, Mandarin  
3                    English, Spanish  
4             English, Hindi, Russian  

[5 rows x 23 columns]

In [101]:
#dropping columns
df_movie_dataset = df_movie_dataset.drop(columns = ['backdrop_path'])

In [102]:
df_movie_dataset = df_movie_dataset.drop(columns = ['poster_path'])

In [103]:
df_movie_dataset = df_movie_dataset.drop(columns = ['overview'])

In [104]:
df_movie_dataset = df_movie_dataset.drop(columns = ['tagline'])

In [105]:
df_movie_dataset = df_movie_dataset.drop(columns = ['homepage'])

In [106]:
df_movie_dataset = df_movie_dataset.drop(columns = ['imdb_id'])

In [107]:
df_movie_dataset = df_movie_dataset.drop(columns = ['original_title'])

In [108]:
df_movie_dataset.head()

id            title  vote_average  vote_count    status release_date  \
0   27205        Inception         8.364       34495  Released   2010-07-15   
1  157336     Interstellar         8.417       32571  Released   2014-11-05   
2     155  The Dark Knight         8.512       30619  Released   2008-07-16   
3   19995           Avatar         7.573       29815  Released   2009-12-15   
4   24428     The Avengers         7.710       29166  Released   2012-04-25   

      revenue  runtime  adult     budget original_language  popularity  \
0   825532764      148  False  160000000                en      83.952   
1   701729206      169  False  165000000                en     140.241   
2  1004558444      152  False  185000000                en     130.643   
3  2923706026      162  False  237000000                en      79.932   
4  1518815515      143  False  220000000                en      98.082   

                                        genres  \
0           Action, Science Fiction, Adventure   
1            Adventure, Drama, Science Fiction   
2               Drama, Action, Crime, Thriller   
3  Action, Adventure, Fantasy, Science Fiction   
4           Science Fiction, Action, Adventure   

                                production_companies  \
0  Legendary Pictures, Syncopy, Warner Bros. Pict...   
1  Legendary Pictures, Syncopy, Lynda Obst Produc...   
2  DC Comics, Legendary Pictures, Syncopy, Isobel...   
3  Dune Entertainment, Lightstorm Entertainment, ...   
4                                     Marvel Studios   

                       production_countries  \
0  United Kingdom, United States of America   
1  United Kingdom, United States of America   
2  United Kingdom, United States of America   
3  United States of America, United Kingdom   
4                  United States of America   

                     spoken_languages  
0  English, French, Japanese, Swahili  
1                             English  
2                   English, Mandarin  
3                    English, Spanish  
4             English, Hindi, Russian

In [109]:
df_movie_dataset.shape

(934745, 16)

In [110]:
#check for missing values
df_movie_dataset.isnull().sum()

id                           0
title                       11
vote_average                 0
vote_count                   0
status                       0
release_date             84528
revenue                      0
runtime                      0
adult                        0
budget                       0
original_language            0
popularity                   0
genres                  328991
production_companies    477871
production_countries    362314
spoken_languages        356693
dtype: int64

In [136]:
df_movie_dataset.describe()

film_id   vote_average     vote_count       revenue  \
count  9.347450e+05  934745.000000  934745.000000  9.347450e+05   
mean   6.223389e+05       2.232957      22.939624  7.782203e+05   
std    3.331466e+05       3.152949     351.094674  1.798041e+07   
min    2.000000e+00       0.000000       0.000000 -1.200000e+01   
25%    3.509120e+05       0.000000       0.000000  0.000000e+00   
50%    6.251880e+05       0.000000       0.000000  0.000000e+00   
75%    9.122350e+05       5.294000       1.000000  0.000000e+00   
max    1.185198e+06      10.000000   34495.000000  2.923706e+09   

             runtime        budget     popularity  
count  934745.000000  9.347450e+05  934745.000000  
mean       51.889123  3.061706e+05       1.424210  
std        62.425681  5.058921e+06       8.349914  
min         0.000000  0.000000e+00       0.000000  
25%         2.000000  0.000000e+00       0.600000  
50%        39.000000  0.000000e+00       0.600000  
75%        90.000000  0.000000e+00       0.958000  
max     14400.000000  8.880000e+08    2994.357000

In [137]:
#check value counts for 'runtime' as the max value was suspicious
df_movie_dataset['runtime'].value_counts()

0      222746
90      27611
10      14497
7       13324
5       13199
        ...  
495         1
885         1
746         1
522         1
605         1
Name: runtime, Length: 701, dtype: int64

In [138]:
#Replace 14400 to 144
df_movie_dataset['revenue'].replace({14400 : 144}, inplace = True)

In [112]:
#check for mixed types
for col in df_movie_dataset.columns.tolist():
  weird = (df_movie_dataset[[col]].applymap(type) != df_movie_dataset[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_movie_dataset[weird]) > 0:
    print (col)

title
release_date
genres
production_companies
production_countries
spoken_languages


In [86]:
#Change data type to string
df_movie_dataset['genres'] = df_movie_dataset['genres'].astype('str')

In [87]:
df_movie_dataset['production_companies'] = df_movie_dataset['production_companies'].astype('str')

In [88]:
df_movie_dataset['production_countries'] = df_movie_dataset['production_countries'].astype('str')

In [89]:
df_movie_dataset['spoken_languages'] = df_movie_dataset['spoken_languages'].astype('str')

In [113]:
df_movie_dataset['title'] = df_movie_dataset['title'].astype('str')

In [114]:
df_movie_dataset['release_date'] = df_movie_dataset['release_date'].astype('str')

In [115]:
# Create a subset, df_nan, to view missing values in a 'genres' column
df_movie_dataset_nan = df_movie_dataset[df_movie_dataset['genres'].isnull() == True]

In [116]:
df_movie_dataset_nan

id                  title  vote_average  vote_count    status  \
6010    320367                 Return         6.223         551  Released   
7684    916035  Barely Legal Baby Fat         6.792         384  Released   
8581     50472   Anplagghed al cinema         7.000         322  Released   
9778    464446                 Return         6.237         264  Released   
10562   292141     The Right of Youth         6.587         235  Released   
...        ...                    ...           ...         ...       ...   
934734  657040        Chasing Trouble         0.000           0  Released   
934735  657042                   City         0.000           0  Released   
934738  657048          Untitled 1985         0.000           0  Released   
934741  657052             The Escape         0.000           0  Released   
934744  657037    Meeting My Daughter         0.000           0  Released   

       release_date  revenue  runtime  adult  budget original_language  \
6010     2015-01-01        0        0  False       0                en   
7684     2008-06-26        0       90   True       0                en   
8581     2006-11-26        0       95  False       0                it   
9778     2015-07-04        0       15  False       0                en   
10562    1911-01-02        0       24  False       0                en   
...             ...      ...      ...    ...     ...               ...   
934734   1926-05-02        0       50  False       0                en   
934735   1985-01-01        0       17  False       0                en   
934738   1985-01-01        0        6  False       0                en   
934741   1926-06-06        0       50  False       0                en   
934744   2001-11-15        0        0  False       0                en   

        popularity genres production_companies      production_countries  \
6010         4.037    NaN                  NaN                       NaN   
7684         5.400    NaN        Hustler Video                       NaN   
8581         4.844    NaN                  NaN                     Italy   
9778         6.767    NaN                  NaN               New Zealand   
10562        5.411    NaN                  NaN                       NaN   
...            ...    ...                  ...                       ...   
934734       0.600    NaN   Universal Pictures  United States of America   
934735       0.600    NaN                  NaN                       NaN   
934738       0.600    NaN                  NaN                       NaN   
934741       0.600    NaN   Universal Pictures  United States of America   
934744       0.600    NaN                  NaN                       NaN   

       spoken_languages  
6010                NaN  
7684            English  
8581            Italian  
9778            English  
10562               NaN  
...                 ...  
934734              NaN  
934735              NaN  
934738              NaN  
934741              NaN  
934744              NaN  

[328991 rows x 16 columns]

In [127]:
# Check for full duplicates
df_movie_dataset_dups = df_movie_dataset[df_movie_dataset.duplicated()]

In [128]:
df_movie_dataset_dups

Empty DataFrame
Columns: [film_id, film_title, vote_average, vote_count, status, release_date, revenue, runtime, adult, budget, original_language, popularity, genres, production_companies, production_countries, spoken_languages]
Index: []

In [129]:
#Rename columns
df_movie_dataset.rename(columns = {'id' : 'film_id'}, inplace = True)

In [130]:
df_movie_dataset.rename(columns = {'title' : 'film_title'}, inplace = True)

In [131]:
#checking data type
df_movie_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 934745 entries, 0 to 934744
Data columns (total 16 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   film_id               934745 non-null  int64  
 1   film_title            934745 non-null  object 
 2   vote_average          934745 non-null  float64
 3   vote_count            934745 non-null  int64  
 4   status                934745 non-null  object 
 5   release_date          934745 non-null  object 
 6   revenue               934745 non-null  int64  
 7   runtime               934745 non-null  int64  
 8   adult                 934745 non-null  bool   
 9   budget                934745 non-null  int64  
 10  original_language     934745 non-null  object 
 11  popularity            934745 non-null  float64
 12  genres                605754 non-null  object 
 13  production_companies  456874 non-null  object 
 14  production_countries  572431 non-null  object 
 15  

In [141]:
df_movie_dataset.head(10)

film_id               film_title  vote_average  vote_count    status  \
0    27205                Inception         8.364       34495  Released   
1   157336             Interstellar         8.417       32571  Released   
2      155          The Dark Knight         8.512       30619  Released   
3    19995                   Avatar         7.573       29815  Released   
4    24428             The Avengers         7.710       29166  Released   
5   293660                 Deadpool         7.606       28894  Released   
6   299536   Avengers: Infinity War         8.255       27713  Released   
7      550               Fight Club         8.438       27238  Released   
8   118340  Guardians of the Galaxy         7.906       26638  Released   
9      680             Pulp Fiction         8.488       25893  Released   

  release_date     revenue  runtime  adult     budget original_language  \
0   2010-07-15   825532764      148  False  160000000                en   
1   2014-11-05   701729206      169  False  165000000                en   
2   2008-07-16  1004558444      152  False  185000000                en   
3   2009-12-15  2923706026      162  False  237000000                en   
4   2012-04-25  1518815515      143  False  220000000                en   
5   2016-02-09   783100000      108  False   58000000                en   
6   2018-04-25  2052415039      149  False  300000000                en   
7   1999-10-15   100853753      139  False   63000000                en   
8   2014-07-30   772776600      121  False  170000000                en   
9   1994-09-10   213900000      154  False    8500000                en   

   popularity                                       genres  \
0      83.952           Action, Science Fiction, Adventure   
1     140.241            Adventure, Drama, Science Fiction   
2     130.643               Drama, Action, Crime, Thriller   
3      79.932  Action, Adventure, Fantasy, Science Fiction   
4      98.082           Science Fiction, Action, Adventure   
5      72.735                    Action, Adventure, Comedy   
6     154.340           Adventure, Action, Science Fiction   
7      69.498                                        Drama   
8      33.255           Action, Science Fiction, Adventure   
9      74.862                              Thriller, Crime   

                                production_companies  \
0  Legendary Pictures, Syncopy, Warner Bros. Pict...   
1  Legendary Pictures, Syncopy, Lynda Obst Produc...   
2  DC Comics, Legendary Pictures, Syncopy, Isobel...   
3  Dune Entertainment, Lightstorm Entertainment, ...   
4                                     Marvel Studios   
5  20th Century Fox, The Donners' Company, Genre ...   
6                                     Marvel Studios   
7  Regency Enterprises, Fox 2000 Pictures, Taurus...   
8                                     Marvel Studios   
9                Miramax, A Band Apart, Jersey Films   

                       production_countries  \
0  United Kingdom, United States of America   
1  United Kingdom, United States of America   
2  United Kingdom, United States of America   
3  United States of America, United Kingdom   
4                  United States of America   
5                  United States of America   
6                  United States of America   
7                  United States of America   
8                  United States of America   
9                  United States of America   

                     spoken_languages  
0  English, French, Japanese, Swahili  
1                             English  
2                   English, Mandarin  
3                    English, Spanish  
4             English, Hindi, Russian  
5                             English  
6                      English, Xhosa  
7                             English  
8                             English  
9            English, Spanish, French

# 04 Saving and Exporting DF

In [143]:
# saving and exporting DF
df_movie_dataset.to_csv(os.path.join(path,'02 Data','Prepared data', 'movie_dataset.csv'))

### Observations of df_contract

The max value for 'runtime' is 14400 min. I believe this is a typo for 144 min, so I changed the value.

There are 11 missing or NaN values for 'title', 84528 for 'release_date', 328991 for 'genres', 477871 for 'production_companies', 362314 for 'production_countries', 356693 for 'spoken_languages'. We need to clarify with the customer which information is important.

Renamed 'id' to 'film_id'
 
Renamed 'title' to 'film_title'
7 columns were dropped 'original_title', 'backdrop_path', 'poster_path', 'overview', 'tagline', 'imdb_id','homepage' 

